### PR TITLE
fix(tabs): fixing null tabs

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Description of changes

There are 2 separate issues here:
1. TS issue where `null` is not allowed as a child of `Tabs`
2. A tangential issue where passing `null` as a child messes up the tab indices: https://codesandbox.io/s/async-flower-772qr1?file=/src/App.js

#### Issue #, if available

#3256

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by reesscot — APPROVED** [review]

File: packages/react/src/primitives/Tabs/Tabs.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ import { Flex } from '../Flex';
 import { TabsProps, TabItemProps, Primitive } from '../types';
 import { View } from '../View';
 
-const isTabsType = (child: any): child is React.Component<TabItemProps> => {
+const isTabsType = (child: any): child is React.ReactElement<TabItemProps> => {
```
Comment (wlee221) [inline packages/react/src/primitives/Tabs/Tabs.tsx:15]: Just curious, why was this change necessary?

**Review by reesscot — DISMISSED** [review]

File: packages/react/src/primitives/Tabs/Tabs.tsx (hunk 1 above)
Comment (zchenwei) [reply packages/react/src/primitives/Tabs/Tabs.tsx:15]: It seems `React.Component` is the legacy class component

File: packages/react/src/primitives/Tabs/__tests__/Tabs.test.tsx [hunk 2]
```diff
@@ -53,6 +53,19 @@ describe('Tabs: ', () => {
...
+  it('should work with defaultIndex and null children', async () => {
+    render(
+      <Tabs testId="tabsTest" defaultIndex={1}>
+        <TabItem title="Tab 1">Tab 1</TabItem>
+        {null}
+        {undefined}
+        <TabItem title="Tab 2">Tab 2</TabItem>
+      </Tabs>
+    );
+    const tabs = await screen.findAllByRole('tab');
```
Comment (wlee221) [inline packages/react/src/primitives/Tabs/__tests__/Tabs.test.tsx:65]: Can we check the number of children to match 2?

File: packages/react/src/primitives/Tabs/Tabs.tsx (hunk 1 above)
Comment (dbanksdesign) [reply packages/react/src/primitives/Tabs/Tabs.tsx:15]: If anyone is curious: https://www.arahansen.com/how-children-types-work-in-react-18-and-typescript-4/

**Review by reesscot — DISMISSED** [review]

**Review by wlee221 — APPROVED** [review]

File: packages/react/src/primitives/Tabs/Tabs.tsx [hunk 3]
```diff
@@ -71,24 +60,34 @@ const TabsPrimitive: Primitive<TabsProps, typeof Flex> = (
...
+        if (isTabsType(child)) {
+          return (
+            <Panel key={index} value={`${index}`}>
+              {child.props.children}
+            </Panel>
+          );
+        } else {
+          // at this point the child defined (not null or undefined)
+          // it is NOT a TabItem, so log a message
+          if (child) {
```
Comment (zchenwei) [inline packages/react/src/primitives/Tabs/Tabs.tsx:84]: nit: could also test this branch.

**Review by zchenwei — APPROVED** [review]

File: packages/react/src/primitives/Tabs/Tabs.tsx (hunk 3 above)
Comment (dbanksdesign) [reply packages/react/src/primitives/Tabs/Tabs.tsx:84]: done!

**Review by wlee221 — DISMISSED** [review]

**Review by wlee221 — APPROVED** [review]

**Review by reesscot — APPROVED** [review]
